tf-modules/distribution/index.js

```diff
@@ -177,6 +178,32 @@ function isPublicRequest(path) {
   }
 }
 
+const isTokenAuthRequest = (req) =>
+  req.get('EDL-Client-Id') && req.get('EDL-Token');
+
+const handleTokenAuthRequest = async (req, res, next) => {
+  const earthdataLoginClient = EarthdataLogin.createFromEnv({
```

Comment:
This duplicates a few lines from getConfigurations() - it might be nice to DRY that into a small helper, use getConfigurations or some other refactor here.